File: 010/010.rb
Primes = Enumerator.new do |yielder|
  yielder << 2
  primes = [2]
  i = 3
  loop do
    unless primes.any? {|p| break if p > Math.sqrt(i) ; i % p == 0 }
      primes << i
      yielder << i
    end
    i += 2
  end
end

puts Primes.take_while {|n| n < 2_000_000 }.inject(&:+)

File: 010/010.clj
(declare primes)

(defn prime? [n]
  (let [factors (take-while #(<= % (Math/sqrt n)) primes)]
    (not-any? #(zero? (mod n %)) factors)))

(def primes
  (lazy-cat [2]
    (filter prime? (take-nth 2 (iterate inc 3)))))

(println (reduce + (take-while #(< % 2000000) primes)))
